hosts/hosts.go

```diff
@@ -28,17 +28,54 @@ const (
 	CleanerImage         = "alpine:latest"
 )
 
-func (h *Host) CleanUp() error {
-	logrus.Infof("[hosts] Cleaning up host [%s]", h.Address)
-	toCleanDirs := []string{
+func (h *Host) CleanUpAll() error {
+	// the only supported removal for etcd dir is in rke remove
+	toCleanPaths := []string{
 		ToCleanEtcdDir,
 		ToCleanSSLDir,
 		ToCleanCNIConf,
 		ToCleanCNIBin,
 		ToCleanCalicoRun,
 	}
+	return h.CleanUp(toCleanPaths)
+}
+
+func (h *Host) CleanUpWorkerHost(controlRole string) error {
+	for _, role := range h.Role {
+		if role == controlRole {
+			logrus.Infof("[hosts] Host [%s] is already a controlplane host, skipping cleanup.", h.Address)
```

Comment:
I'd rephrase this message to something like: `Worker host is a control place host..`